lib/pipeline.js

```diff
@@ -301,6 +303,47 @@ class PipelineModel extends BaseModel {
                         throw new Error(message);
                     });
                 })
+                .then(screwdriverConf => {
+                    if (pipelineFactory.isExtendingEnabled()) {
```

Comment:
this should based on a specific individual pipeline, if an organization wants to enforce this for some but not all.

Reply:
that can be again as per the org decision. In my case, if we are adopting, we want all the repo to follow that, and in case any repo wants to try something else they could choose it by overriding it at repo level. We should provide both. If the user don't want global override, ignore the config, else add a config. Again, give freedom to the user's needs. a good product always needs flexibility to adoption. Take an example of Jira. along with defining an std process to follow, they have given away to override each and everything, and that's the single reason for the enterprise adoption of the solution.

Reply:
my feedback is around if user provides a complete `screwdriver.yaml` use it instead of always calling external API even if it's enabled.

Are you looking at more of a lock-in per SD instance, where all users must use pipeline config from this external API ?